enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1031 @@
+
+For example: If someone creates a Ingress Controller with the following:
+```sh
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  httpHeaders:
+    forwardedHeaderPolicy: Replace
+```
+Then the `set-header` stanzas will be added in `haproxy.config` using the [haproxy-config.template](https://github.com/openshift/router/blob/0cd9a58dff77127562f7851ccfa5cda79b8cef48/images/router/haproxy/conf/haproxy-config.template#L565-L580)
+However, if someone wants to set custom values for these then they need to use the API of this EP.
+
+If both are set then custom values will take over.  
+
+Please note that the setting of forwardedHeaderPolicy happens at the plain backend section of haproxy.config i.e. per route level and not at the front-end section
+of the haproxy.config level as per the above-mentioned link.
+
+Another use case is deleting the header which forwardedHeaderPolicy does not provide but this EP does.
+
+#### X-SSL
+Similarly to `x-forwarded-for`, there are legitimate use-cases for setting `x-ssl` and related headers, and allowing the cluster-admin or route owner to modify these headers presents no security risk.
+For example: HAProxy takes information from TLS handshake, tls session or the certificate that client provided. These headers are used for mtls and client certificates.
+
+These headers tell you whether the client provided a certificate, was the certificate valid, what was the content or cipher, distinguished name or expiration date
+on the certificate using converters with prefix like `ssl_` which can be used in the haproxy proxy config to take that information and put it http request headers which get sent
+to the backend application.
+
+You must already trust the cluster-admin or route owner, who has full control over the proxy or application.  Thus, there is no additional risk in trusting
+an `x-ssl` header that is controlled by the cluster-admin or route owner.  For this reason, we do not restrict setting this header.
+
+#### Host
+Setting the `host` header via the IngressController CR can cause HAProxy to fail to look up the correct route, and for that reason, we prohibit setting the `host` request header on the IngressController.
+
+OpenShift router's `haproxy.config` file uses [the `base` sample fetch method](http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#7.3.6-base) in the frontends to get the
+HTTP request `host` header and path, which are then used with [the `map` converter](http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#7.3.1-map) to look up the
+appropriate backend to use for the request.
+
+Modifying the `host` header on the IngressController therefore could cause HTTP requests to fail with HTTP 503 errors.
+For this reason, the `host` header is in the deny list of headers that are prohibited to be set or deleted using the IngressController API.
+
+### Drawbacks
```

Comment:
I would list the potential difficulty in documenting and troubleshooting this as a drawback, e.g. all the special treatment of certain headers.

Reply:
i have pointed this section to `Risks and Mitigation` section